sycl/include/sycl/ext/intel/esimd/detail/atomic_intrin.hpp

```diff
@@ -30,52 +39,87 @@ template <typename Ty> Ty atomic_store(Ty *ptr, Ty val) {
   // TODO: Windows will be supported soon
   __ESIMD_UNSUPPORTED_ON_HOST;
 #else
-  __atomic_store(ptr, val, __ATOMIC_SEQ_CST);
+  Ty ret = atomic_load<Ty>((CmpxchgTy<Ty> *)ptr);
```

Comment:
it looks strange that we need atomic_load as a part of atomic_store implementation, but that's how Gen ISA is :)
no action for this comment is needed

Reply:
I asked Victor Mustya for clarification.